src/Microsoft.Graph.Core/Requests/Content/BatchResponseContent.cs

```diff
@@ -55,10 +60,10 @@ public BatchResponseContent(HttpResponseMessage httpResponseMessage, ISerializer
             if (jBatchResponseObject == null)
                 return responseMessages;
 
-            if(jBatchResponseObject.TryGetValue(CoreConstants.BatchRequest.Responses, out JToken jResponses))
+            if(jBatchResponseObject.RootElement.TryGetProperty(CoreConstants.BatchRequest.Responses, out JsonElement jResponses))
```

Comment:
```suggestion
            if(jBatchResponseObject.RootElement.TryGetProperty(CoreConstants.BatchRequest.Responses, out JsonElement jResponses) && jResponses != null)
```

Reply:
probably need to test whether it's an array as well actually

Reply:
JsonElement is actually a struct and is therefore not nullable. Will resolve with the check `jResponses.ValueKind == JsonValueKind.Array`